### ISSUE TITLE
Reference key format 

### ISSUE THREAD

**Issue by nealsid** [bigdatagenomics/adam#128]
We've been using integer ids for reference keys - we thought that using integer reference/contig ids was better because it prevented naming conflicts when processing multiple BAM or VCF files, since it's essentially uniquifying the sequence dictionary keys.  However, we can also do this by using RefSeq strings as identifiers, including the reference version information.  If we're processing multiple BAM or alignment files and they specify both specify the same random (i.e. non RefSeq) contig name that could mean different contigs for each BAM file, it's a problem.  However, it is fairly rare, and we can detect/throw an error/punt on it for now.

Also, the region join work is blocked on this; as Timothy summarized in issue 127:
- Neal is going to make the change from using refId -> referenceName as the primary key, and will break it out as a separate PR
- the regionJoin PR will be blocked on the merging of that PR first
- it will include a way to specify a mapping from "common contig name" to a unique ("full Refseq ID") name at load time, for both BAMs and VCFs.
- Michael will add liftover files that specify those conversions for at least builds 37 and 38 for human
